components/gitpod-protocol/src/workspace-cluster.ts

```diff
@@ -9,6 +9,12 @@ import { filePathTelepresenceAware } from "./env";
 import { DeepPartial } from "./util/deep-partial";
 import { PermissionName } from "./permission";
 
+export type WorkspaceRegion = "europe" | "north-america" | "south-america" | "africa" | "asia" | ""; // unknown;
+
+export function isWorkspaceRegion(s: string): s is WorkspaceRegion {
+    return ["europe", "north-america", "south-america", "africa", "asia", ""].indexOf(s) !== -1;
+}
```

Comment:
In case you don't want to repeat the list:
```ts
const workspaceRegions = ["europe", "north-america", "south-america", "africa", "asia", ""] as const;
export type WorkspaceRegion = typeof workspaceRegions[number];

export function isWorkspaceRegion(s: string): s is WorkspaceRegion {
    return workspaceRegions.indexOf(s as any) !== -1;
}
```

Reply:
How does the `[number]` part work?

Reply:
I just gives you the type of any `workspaceRegions[number]` expression. I.e. `workspaceRegions[0]` would be of type "Europe".

Reply:
Interesting, hadn't seen this syntax before. And the `[number]` doesn't feel intuitive but perhaps just my lack of being used to it. Thanks.